snapcraft/tests/sources/test_checksum.py

```diff
@@ -0,0 +1,85 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from snapcraft import tests
+from snapcraft.internal import sources
+
+import os
+import hashlib
+import zipfile
+import sys
+
+if sys.version_info < (3, 6):
+    import sha3  # noqa
+
+
+class TestChecksum(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+
+    def test_invalid_checksum(self):
+        # Create zip file for testing
+        os.makedirs(os.path.join('src'))
+        file_to_zip = os.path.join('src', 'test.txt')
+        open(file_to_zip, 'w').close()
+        zip_file = zipfile.ZipFile(os.path.join('src', 'test.zip'), 'w')
+        zip_file.write(file_to_zip)
+        zip_file.close()
```

Comment:
on this one, there's no need to make a zip file, right?
The error should happen before verify_checksum tries to open the file, so you can just pass a string like 'dummy.zip', and you test the same code path faster.